service/raft/raft_group0.cc

```diff
@@ -133,31 +133,43 @@ raft_group0::raft_group0(seastar::abort_source& abort_source,
     : _abort_source(abort_source), _raft_gr(raft_gr), _ms(ms), _gossiper(gs), _qp(qp), _mm(mm), _feat(feat), _sys_ks(sys_ks), _client(client)
     , _status_for_monitoring(_raft_gr.is_enabled() ? status_for_monitoring::normal : status_for_monitoring::disabled)
 {
-    init_rpc_verbs();
     register_metrics();
 }
 
+future<> raft_group0::start() {
+    return smp::invoke_on_all([this]() {
+        init_rpc_verbs();
+    });
+}
+
 void raft_group0::init_rpc_verbs() {
-    ser::group0_rpc_verbs::register_group0_peer_exchange(&_ms, [this] (const rpc::client_info&, rpc::opt_time_point, discovery::peer_list peers) {
-        return peer_exchange(std::move(peers));
+    ser::group0_rpc_verbs::register_group0_peer_exchange(&_ms.local(), [this] (const rpc::client_info&, rpc::opt_time_point, discovery::peer_list peers) {
```

Comment:
Without RAII for the registration this is not exception safe. If registration fails during raft_group0::start(), already registered verbs will not be unregistered.

Reply:
It may well be a preexisting problem, but we should follow up on this.